Task 7/Task 7.2/script.js

```diff
@@ -0,0 +1,115 @@
+calculateExpression();
+
+function calculateExpression () {
+    try {
+        let userInput = document.getElementById("userExpression");
+        let output = document.getElementById("result");
+        userInput.addEventListener("blur", (event) => {
+            let result = calculator (event.target.value);
+            output.innerHTML = "Your result is " + result;
+        });
+    } catch (e) {
+        alert(e.message);
+    }
```

Comment:
You register a callback which will be called when the event is fired. The registering part does and waits nothing, because it doesn't call the callback, so catching an error will do nothing here, unless there's an error during setup. To actually catch errors that happen during event processing, you should catch errors there.